rewrite-maven/src/main/java/org/openrewrite/maven/AddOrUpdateDependencyManagement.java

```diff
@@ -0,0 +1,291 @@
+                                emptyList(),
+                                emptyList(),
+                                new Xml.Tag.Closing(randomId(), "dependencyManagement", "", fmt1),
+                                "",
+                                fmt1
+                        )
+                ));
+            }
+            // Add a <dependencies> tag beneath <dependencyManagement> if one does not already exist
+            boolean hasDependenciesTag = p.getDependencyManagement()
+                    .getTag()
+                    .getChildren()
+                    .stream()
+                    .anyMatch(it -> it.getName().equals("dependencies"));
+            if (!hasDependenciesTag) {
+                Maven.DependencyManagement currentDepMan = p.getDependencyManagement();
+                Xml.Tag tag = currentDepMan.getTag();
+                List<Content> contents = new ArrayList<>(tag.getContent());
+                Xml.Tag dependenciesTag = new Xml.Tag(
+                        randomId(),
+                        "dependencies",
+                        emptyList(),
+                        emptyList(),
+                        new Xml.Tag.Closing(randomId(), "dependencies", "", fmt2),
+                        "",
+                        fmt2);
+                contents.add(dependenciesTag);
+                currentDepMan = new Maven.DependencyManagement(currentDepMan.getModel(), tag.withContent(contents));
+                p = p.withDependencyManagement(currentDepMan);
+            }
+            return p;
+        }
+    }
+
+    private class InsertDependencyInOrder extends MavenRefactorVisitor {
+        @Override
+        public Maven visitPom(Maven.Pom pom) {
+            Maven.Pom p = refactor(pom, super::visitPom);
+            List<Maven.Dependency> dependencies = new ArrayList<>(pom.getDependencyManagement().getDependencies());
+
+            // TODO if the dependency is manageable, make it managed
+            Xml.Tag dependencyTag = createDependencyTag(formatter.wholeSourceIndent());
+
+            Maven.Dependency toAdd = new Maven.DependencyManagement.Dependency(false,
+                    new MavenModel.Dependency(
+                            new MavenModel.ModuleVersionId(groupId, artifactId, null, version, "jar"),
+                            version,
+                            scope
+                    ),
+                    dependencyTag
+            );
+
+            if (dependencies.isEmpty()) {
+                dependencies.add(toAdd);
+            } else {
+                // if everything were ideally sorted, which dependency would the addable dependency
+                // come after?
+                List<Maven.Dependency> sortedDependencies = new ArrayList<>(pom.getDependencyManagement().getDependencies());
+                sortedDependencies.add(toAdd);
+                dependencies.sort(Comparator.comparing(d -> d.getModel().getModuleVersion()));
```

Comment:
This looks like the version number is the only sorting criteria. Am I reading that correctly?
I would expect that it would probably sort first by group, then by artifact.

Reply:
Yes, seems correct. But that's the unchanged copy from the AddDependency class we modified.

Reply:
Should I change it here ?